django-tethne/models.py

```diff
@@ -0,0 +1,163 @@
+This refers to the Paper instance(Foreign Key) and thus represents a record in the corpus.
+
+"""
+
+class Author_Instance(models.Model):
+    paper = models.ForeignKey(Paper)
+    author = models.ForeignKey(Author)
+    au_fname = models.CharField(max_length=45, null=True)
+    au_lname = models.CharField(max_length=45, null=True)
+
+
+
+"""
+
+Model name : Author_Identity
+This represents the relation Author <-> Author_identity <-> Author_instance.
+An additional field confidence is added in this relation.
+
+"""
+
+class Author_Identity(models.Model):
+    author = models.ForeignKey(Author)
+    author_instance = models.ForeignKey(Author_Instance)
+    confidence = models.FloatField(null=True)
+
+
+
+"""
+
+Model Name : Institution
+
+Will store all the details of a specific institution
+
+"""
+
+class Institution(models.Model):
+    name = models.CharField(max_length=45, null=True)
+    addressLine1 = models.CharField(max_length=50, null=True)
+    addressLine2 = models.CharField(max_length=50, null=True)
+    state = models.CharField(max_length=45, null=True)
+    city = models.CharField(max_length=45, null=True)
+    zip = models.CharField(max_length=45, null=True)
+    country = models.CharField(max_length=45, null=True)
+
+
+"""
+Model Name : Institution_Instance
+
+Will store a record that appears in the Corpus.
+Thus refers to the Paper.
+"""
+
+class Institution_Instance(models.Model):
+    institution = models.ForeignKey(Institution)
+    paper = models.ForeignKey(Paper)
+    department = models.CharField(max_length=45, null=True)
+    name = models.CharField(max_length=45, null=True)
+    addressLine1 = models.CharField(max_length=50, null=True)
+    addressLine2 = models.CharField(max_length=50, null=True)
+    state = models.CharField(max_length=45, null=True)
+    city = models.CharField(max_length=45, null=True)
+    zip = models.CharField(max_length=45, null=True)
+    country = models.CharField(max_length=45, null=True)
+
+
+
+
+
+class Institution_Identity(models.Model):
+    institution = models.ForeignKey(Institution)
+    institution_instance = models.ForeignKey(Institution_Instance)
+    confidence = models.FloatField(null=True)
+
+
+
+class Affiliation(models.Model):
+    author = models.ForeignKey(Author)
+    institution = models.ForeignKey(Institution)
+    start_date = models.DateField(null=True)
+    end_date = models.DateField(null=True)
+    confidence = models.FloatField(null=True)
+
+
+class Affiliation_instance(models.Model):
+    author_instance = models.ForeignKey(Author_Instance)
+    institution_instance = models.ForeignKey(Institution_Instance)
+
+
+class Citation_Instance(models.Model):
+    paper = models.ForeignKey(Paper)
+    literal = models.CharField(max_length=45, null=True)
+    journal = models.CharField(max_length=45, null=True)
+    first_author = models.CharField(max_length=45, null=True)
+    date = models.DateField(null=True)
+
+
+class Citation_Indentity(models.Model):
```

Comment:
This should also point at ``Citation`` via FK (an instance of this class represents the assertion that a particular cited reference (``Citation_Instance``) refers to a specific ``Citation``.